src/main/java/gg/jte/demo/TemplateRenderer.java

```diff
@@ -16,13 +17,13 @@
 @Service
 public class TemplateRenderer {
 
-    // TODO determine if we run spring boot in development mode or not
-    private final boolean devMode = false;
+    @Value("${spring.profiles.active}")
+    private String profile;
 
     private final TemplateEngine templateEngine;
 
     public TemplateRenderer() {
-        if (devMode) {
+        if (profile != "prod") {
```

Comment:
You should use `"prod".equals(profile)` here. It's very likely that the injected spring profile is not the same reference as the literal in the constant pool here.

Reply:
That was it, thanks! :facepalm: 

I've messed up my git history, I'll play with this some more to make sure it's stable, the will reopen or update the PR.